src/main/java/de/medizininformatikinitiative/torch/ConsentHandler.java

```diff
@@ -195,6 +199,94 @@ public Flux<Map<String, Map<String, List<ConsentPeriod>>>> buildingConsentInfo(S
 
 
 
+    public Flux<Map<String, Map<String, List<ConsentPeriod>>>> updateConsentInfo(
+            Flux<Map<String, Map<String, List<ConsentPeriod>>>> consentInfoFlux, List<String> batch) {
+
+        logger.info("Starting to update consent info with batch size: {}", batch.size());
+
+        // Step 1: Fetch all encounters for the batch of patients
+        Flux<Encounter> allEncountersFlux = dataStore.getResources("Encounter", FhirSearchBuilder.getEncounter(batch))
+                .subscribeOn(Schedulers.boundedElastic())
+                .cast(Encounter.class)
+                .doOnSubscribe(subscription -> logger.info("Fetching encounters for batch: {}", batch))
+                .doOnNext(encounter -> logger.debug("Encounter fetched: {}", encounter.getIdElement().getIdPart()))
+                .onErrorResume(e -> {
+                    logger.error("Error fetching encounters for batch: {}", batch, e);
+                    return Flux.empty();
+                });
+
+        // Step 2: Group the encounters by patient ID
+        Mono<Map<String, Collection<Encounter>>> encountersByPatientMono = allEncountersFlux
+                .flatMap(encounter -> {
+                    try {
+                        String patientId = ResourceUtils.getPatientId(encounter);
+                        // Use Map.Entry to pair patient ID with the encounter
+                        return Mono.just(new AbstractMap.SimpleEntry<>(patientId, encounter));
+                    } catch (PatientIdNotFoundException e) {
+                        logger.error("Patient ID not found in encounter resource", e);
+                        return Mono.empty(); // Skip this encounter
+                    }
+                })
+                .collectMultimap(
+                        Map.Entry::getKey, // Key Mapper: patient ID
+                        Map.Entry::getValue // Value Mapper: Encounter object
+                );
+
+        // Step 3: Process each patient's consent info individually
+        return encountersByPatientMono.flatMapMany(encountersByPatientMap ->
```

Comment:
@Lucas0T why not extract the current patients consent set und pass that on to updateConsentPeriodsForPatient
and then return updateConsentPeriodsForPatient()